libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Steps/HttpRequest.cs

```diff
@@ -132,8 +137,19 @@ private async Task ReplaceJTokenRecursively(DialogContext dc, JToken token)
                 default:
                     if (token.Type == JTokenType.String)
                     {
-                        token.Replace(await new TextTemplate(token.ToString()).BindToData(dc.Context, dc.State));
+                        var temp = await new TextTemplate(token.ToString()).BindToData(dc.Context, dc.State);
+                        if ((temp.StartsWith("{") && temp.EndsWith("}")) || (temp.StartsWith("[") && temp.EndsWith("]")))
+                        {
+                            // try parse with json                        
+                            var jtoken = JToken.Parse(temp);
+                            token.Replace(jtoken);
+                        }
+                        else
+                        {
+                            token.Replace(temp);
+                        }
```

Comment:
What is this?  Why do you need this?

Reply:
previously we just replace the temp object, but it will not handle this case: if the temp is a JArray or JObject, the result will have double quotations around the JArray which is an invalid json. So for any json objects in memory, we check the type firstly and try to parse with jtoken and replace.

Reply:
this change is to solve #2315